Neos.Cache/Classes/Frontend/AbstractFrontend.php

```diff
@@ -152,6 +152,27 @@ public function flushByTag(string $tag): int
         return 0;
     }
 
+    /**
+     * Removes all cache entries of this cache which are tagged by any of the specified tags.
+     *
+     * @param array<string> $tags The tags the entries must have
+     * @return integer The number of entries which have been affected by this flush
+     * @throws \InvalidArgumentException
+     * @api
+     */
+    public function flushByTags(array $tags): int
+    {
+        foreach ($tags as $tag) {
+            if (!$this->isValidTag($tag)) {
+                throw new \InvalidArgumentException('"' . $tag . '" is not a valid tag for a cache entry.', 1646209443);
+            }
+        }
```

Comment:
I actually also wondered why we need this. If the tag is invalid it wouldn't hurt anyway.

Maybe there is a use case to prevent SQL injections or something?
But then I agree too, that the backend should take care of that.

Reply:
It might even be that a tag that is valid according to the frontend is dangerous in the backend (it should not, but it _might_) so IMO this _has_ to be tested/sanitized in the backend anyways

Reply:
Agree. But should I really do this in this PR too?

Reply:
There is no need to be smarter than the current solution. As this is done already in flushByTag it does not produce extra costs we do not already pay.

Reply:
> There is no need to be smarter than the current solution

I beg to differ: We introduce this method exclusively for performance reasons, so IMO we can decide to be smarter than before :)
But I was expecting our backends to validate the tag and not all of them do.. So this would be too much of a change for one PR I guess

Reply:
I can prepare a 2nd PR for moving the validation to the backends.